packages/components/src/components/CollapsibleBox/CollapsibleBox.tsx

```diff
@@ -184,10 +193,15 @@ const CollapsibleBox: FC<CollapsibleBoxProps> & CollapsibleBoxSubcomponents = ({
         setIsCollapsed(!isOpen);
     }, [isOpen]);
 
-    const handleHeaderClick = useCallback(() => {
-        onCollapse?.();
-        setIsCollapsed(!isCollapsed);
-    }, [isCollapsed, onCollapse]);
+    const handleHeaderClick = useCallback(
+        (e: MouseEvent<HTMLDivElement>) => {
+            onToggle?.(!!isCollapsed); // !!, one for change of the state, the other for isCollapsed => isOpen
```

Comment:
- make isOpen optional, 
- create managed and unmanaged mode for this component, 
- unify it as isOpen (`isCollapsed` is negative and unintuitive in combination with isOpen)